ksql-plugins/ksql-rocksdb-config-setter/src/main/java/io/confluent/ksql/rocksdb/KsqlBoundedMemoryRocksDBConfig.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rocksdb;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.kafka.streams.state.RocksDBConfigSetter;
+import org.rocksdb.BlockBasedTableConfig;
+import org.rocksdb.Options;
+
+/**
+ * This {@code RocksDBConfigSetter} implementation limits the total memory used
+ * across all RocksDB instances to the number of bytes passed via
+ * "ksql.plugins.rocksdb.total.memory", and also configures
+ * the shared RocksDB thread pool to use "ksql.plugins.rocksdb.num.background.threads" threads.
+ * </p>
+ * See https://docs.confluent.io/5.3.0/streams/developer-guide/memory-mgmt.html#rocksdb.
+ */
+public class KsqlBoundedMemoryRocksDBConfig implements RocksDBConfigSetter {
+
+  private static final String CONFIG_PREFIX = "ksql.plugins.rocksdb.";
+  private static final String TOTAL_OFF_HEAP_MEMORY_CONFIG = CONFIG_PREFIX + "total.memory";
+  private static final String N_BACKGROUND_THREADS_CONFIG =
+      CONFIG_PREFIX + "num.background.threads";
+
+  private static final double INDEX_FILTER_BLOCK_RATIO = 0.1;
```

Comment:
For some but not all.

- `INDEX_FILTER_BLOCK_RATIO` defaults to 0.0 (in RocksDB, and Kafka Streams uses this default). 

- The default RocksDBConfigSetter used by Streams sets 3 for `N_MEMTABLES`, though @ableegoldman 's performance testing suggests that 6 tends to yield better results.

- `BLOCK_SIZE` of 4096L aligns with the value set by Streams's default RocksDBConfigSetter.

Reply:
Sounds like we should expose these as configurable, rather than hardcoding, and they should default to whatever streams defaults to, (ideally picking up the defaults from the streams code, rather than duplicating it, which can diverge).